src/mscorlib/src/System/Reflection/AssemblyNameHelpers.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*============================================================
+**
+  Type:  AssemblyNameHelpers
+**
+==============================================================*/
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    internal static class AssemblyNameHelpers
+    {
+        private const int PUBLIC_KEY_TOKEN_LEN = 8;
+
+        public static String ComputeDisplayName(AssemblyName a)
+        {
+            if (a.Name == String.Empty)
+                throw new FileLoadException();
+
+            StringBuilder sb = new StringBuilder();
+            if (a.Name != null)
+            {
+                sb.AppendQuoted(a.Name);
+            }
+
+            if (a.Version != null)
+            {
+                Version canonicalizedVersion = a.Version.CanonicalizeVersion();
+                if (canonicalizedVersion.Major != ushort.MaxValue)
+                {
+                    sb.Append(", Version=");
+                    sb.Append(canonicalizedVersion.Major);
+                    sb.Append('.');
+                    sb.Append(canonicalizedVersion.Minor);
```

Comment:
Actually, `sb.Append(a.Version.ToString())` may work just fine instead all this.